packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -1202,7 +1205,7 @@ class TextStyle with Diagnosticable {
         forceStrutHeight: strutStyle.forceStrutHeight,
       ),
       maxLines: maxLines,
-      ellipsis: ellipsis,
+      ellipsis: overflow == TextOverflow.ellipsis ? (ellipsis ?? kDefaultEllipsis) : null,
```

Comment:
The local `ellipsis` should have the highest priority and override the value from the class I think?

```suggestion
      ellipsis: ellipsis ?? (overflow == TextOverflow.ellipsis ? kDefaultEllipsis : null),
```

Reply:
Wow, this was a tricky one. Thank you!